### PR TITLE
Add ADR on release package signing 

### PR DESCRIPTION
**Pull request by toban**

### CODE REVIEW

File: docs/adr/0014-signing-release-packages.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Date: 2021-05-03
+
+## Status
+
+proposed
+
+## Context
+
+During the course of the first hike of the release strategy[T272418](https://phabricator.wikimedia.org/T272418) a brief investigation into signing of release packages was done.
```
Comment (mariushoch) [inline docs/adr/0014-signing-release-packages.md:11]: The link seems out of place… also aren't we the second hike?

File: docs/adr/0014-signing-release-packages.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- Tarball packages
+- Docker images
+
+### Signing tarball packages with GPG
+
+Tarball files can be signed with [GPG](https://gnupg.org/gph/en/manual/x135.html) in a similar way to which MediaWiki are signing theirs. In the case of MediaWiki a private and public key-pair is generated for release engineering members and published on https://www.mediawiki.org/keys/keys.html. 
+
+By importing the list of public keys the end-user can then verify the release tarball and it's signature by issuing the following command.
+
+```
```
Comment (mariushoch) [inline docs/adr/0014-signing-release-packages.md:24]: Maybe mention that `gpg --fetch-keys "https://www.mediawiki.org/keys/keys.txt"` needs to run first?

File: docs/adr/0014-signing-release-packages.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+#### Signer key
+
+The signer key is what is added to the docker registry and can be delegated in such a way that more than one signer can sign images for a single repository.
+
+### Managing DCT and GPG keys
+
+For DCT the signer and the publisher does not necessarily have to be the same person. A image can be published and then the signer can add a signature later by pulling and signing. (This way managing keys does not have to be a burden on everyone).
+
+Root keys and GPG keys that represent WMDE as an organization should be generated and kept in a secure location. As there has been no previous singing of release packages the methods and processes required for maintaining, using and backing up these keys would have to be defined and formalized before they are applied.
```
Comment (mariushoch) [inline docs/adr/0014-signing-release-packages.md:57]: s/singing/signing/

File: docs/adr/0014-signing-release-packages.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Therefore the decision for the release pipeline will be not to sign any of the release packages in the first version (wmde.0 and wmde.1).
+
+However the benefits of signing are obvious and therefore it's probably a good idea to take the time and answer the following questions to have a solid foundation to stand on when doing release signing.
+
+- Which engineers are to be considered release engineers?
+    - Needs access to organization in docker registry
+    - Needs access to organization in github
+    - Need access to `releasers-wikibase` group
+    - Would need to publish and maintain a GPG key
+    - Would potentially need to maintain a DCT signer key and a public version unless the GPG key can be re-used.
```
Comment (mariushoch) [inline docs/adr/0014-signing-release-packages.md:68]: According to the DCT documentation it should be possible to use the same key (`docker trust key load`).

File: docs/adr/0014-signing-release-packages.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+- How do we securely store/generate offline "organization" private keys?
+
+- Where do WMDE publish public keys?
```
Comment (mariushoch) [inline docs/adr/0014-signing-release-packages.md:72]: s/do/does/

File: docs/adr/0014-signing-release-packages.md (hunk 2 above)
Comment (toban) [reply docs/adr/0014-signing-release-packages.md:24]: done

File: docs/adr/0014-signing-release-packages.md (hunk 3 above)
Comment (toban) [reply docs/adr/0014-signing-release-packages.md:57]: done

File: docs/adr/0014-signing-release-packages.md (hunk 4 above)
Comment (toban) [reply docs/adr/0014-signing-release-packages.md:68]: done

File: docs/adr/0014-signing-release-packages.md (hunk 5 above)
Comment (toban) [reply docs/adr/0014-signing-release-packages.md:72]: done

**Pull request merged**